pulsar-io/aerospike/src/main/java/org/apache/pulsar/io/aerospike/AerospikeSinkConfig.java

```diff
@@ -38,21 +42,50 @@ public class AerospikeSinkConfig implements Serializable {
     private String columnName;
 
     // Optional
+    @FieldDoc(
+            required = false,
+            defaultValue = "",
+            sensitive = true,
+            help = "The username for authentication."
+    )
     private String userName;
+    @FieldDoc(
+            required = false,
+            defaultValue = "",
+            sensitive = true,
+            help = "The password for authentication."
+    )
     private String password;
     private String keySet;
     private int maxConcurrentRequests = 100;
     private int timeoutMs = 100;
     private int retries = 1;
 
 
+    /**
+     * @deprecated Use {@link #load(String, SinkContext)} instead.
+     */
+    @Deprecated
     public static AerospikeSinkConfig load(String yamlFile) throws IOException {
         ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
         return mapper.readValue(new File(yamlFile), AerospikeSinkConfig.class);
     }
 
+    public static AerospikeSinkConfig load(String yamlFile, SinkContext context) throws IOException {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        return load(mapper.readValue(new File(yamlFile), new TypeReference<Map<String, Object>>() {}), context);
+    }
+
+    /**
+     * @deprecated Use {@link #load(Map, SinkContext)} instead.
+     */
+    @Deprecated
     public static AerospikeSinkConfig load(Map<String, Object> map) throws IOException {
         ObjectMapper mapper = new ObjectMapper();
         return mapper.readValue(mapper.writeValueAsString(map), AerospikeSinkConfig.class);
     }
+
+    public static AerospikeSinkConfig load(Map<String, Object> map, SinkContext context) {
```

Comment:
We should just keep the method, all other `load` methods should be removed, and the relevant unit tests should be fixed.

Reply:
From the `yamlFile` load config is useless. This is not the responsibility of each connector but the responsibility of the connector framework.

Reply:
> We should just keep the method, all other `load` methods should be removed, and the relevant unit tests should be fixed.

I am conflicted about removing those `public` methods. I agree this class is cluttered, but it introduces an unnecessary breaking change to outright remove them.

> From the `yamlFile` load config is useless. This is not the responsibility of each connector but the responsibility of the connector framework.

What exactly is the connector framework's responsibility?

Reply:
> I am conflicted about removing those public methods. I agree this class is cluttered, but it introduces an unnecessary breaking change to outright remove them.

For each connector, no user to use its class, right? We just need to make sure that the configuration is compatible.

I don't think removing these public methods would be a breaking change.

> What exactly is the connector framework's responsibility?
The io framework is responsible for loading the configuration from the YAML and converting it to the Map. So for the connector, we only need cover to deserialize the correct configuration from the Map.

Reply:
> For each connector, no user to use its class, right? We just need to make sure that the configuration is compatible.

I think you're right here. Part of the issue is likely the ambiguous boundaries for users. It's probably possible to use this class as a dependency in another project, but I doubt there is really an expectation that these classes are meant to be used elsewhere. The primary risk would be for someone slightly modifying a connector.

> The io framework is responsible for loading the configuration from the YAML and converting it to the Map. So for the connector, we only need cover to deserialize the correct configuration from the Map.

As long as we assume these classes are for pulsar's internal use, I agree that this is the framework's responsibility.

The last question is really whether we think this change is valuable. @eolivelli is asking in another comment on this PR whether this is the right design.

Reply:
> I think you're right here. Part of the issue is likely the ambiguous boundaries for users. It's probably possible to use this class as a dependency in another project, but I doubt there is really an expectation that these classes are meant to be used elsewhere. The primary risk would be for someone slightly modifying a connector.

Maybe we can discuss clearly where the boundaries we offer to users are. I don't think users should use this class as a dependency in another project. We shouldn't have uploaded the connector project to the Maven center, right?


> The last question is really whether we think this change is valuable. @eolivelli is asking in another comment on this PR whether this is the right design.

Maybe the IO framework needs to deserialize the configuration object for the connector. 

```
public interface Source<C, T> extends AutoCloseable {

    /**
     * Open connector with configuration.
     *
     * @param config initialization config
     * @param sourceContext environment where the source connector is running
     * @throws Exception IO type exceptions when opening a connector
     */
    void open(C config, SourceContext sourceContext) throws Exception;

    /**
     * Reads the next message from source.
     * If source does not have any new messages, this call should block.
     * @return next message from source.  The return result should never be null
     * @throws Exception
     */
    Record<T> read() throws Exception;
}
```